src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+
+    this->picto.actor.focus.pos = this->picto.actor.world.pos;
+    this->picto.actor.focus.pos.y = this->picto.actor.world.pos.y + 100.0f;
+
+    this->actionFunc = EnBal_FloatIdle;
+}
+
+void EnBal_FloatIdle(EnBal* this, PlayState* play) {
+    f32 scaleFactor;
+
+    this->skyFloatPhase += 800;
+    this->picto.actor.focus.pos.y = this->picto.actor.world.pos.y + 100.0f;
+    this->timer += 1000;
+
+    scaleFactor = (Math_CosS(this->skyFloatPhase) * 0.1f) + 1.0f;
+    this->balloonScale.z = scaleFactor;
+    this->balloonScale.y = scaleFactor;
+
+    scaleFactor = (Math_SinS(this->skyFloatPhase) * 0.1f) + 1.0f;
+    this->balloonScale.x = SQ(scaleFactor);
+
+    this->picto.actor.world.pos.y = (Math_SinS(this->skyFloatPhase) * 50.0f) + this->picto.actor.home.pos.y;
+    if (this->forceEyesShut == true) {
+        if (this->skelAnime.animation != &gTingleHideFaceAnim) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_HIDE_FACE);
+        } else if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            if (this->idleAnimStage != EN_BAL_IDLESTAGE_ACTIVITY) {
+                this->idleAnimStage--;
+            } else {
+                this->forceEyesShut = false;
+                this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_FLOAT_DRAW);
+            }
+        }
+    }
+}
+
+void EnBal_SetupPopBalloon(EnBal* this) {
+    EnBal_SetMainColliderToHead(this);
+    this->timer = 0;
+    Actor_PlaySfx(&this->picto.actor, NA_SE_EV_MUJURA_BALLOON_BROKEN);
+    this->prevBalloonAction = EN_BAL_BALACT_POP;
+    this->actionFunc = EnBal_PopBalloon;
+}
+
+void EnBal_PopBalloon(EnBal* this, PlayState* play) {
+    this->picto.actor.focus.pos.x = this->picto.actor.world.pos.x;
+    this->picto.actor.focus.pos.y = this->picto.actor.world.pos.y + 20.0f;
+    this->picto.actor.focus.pos.z = this->picto.actor.world.pos.z + 30.0f;
+    if (this->picto.actor.csId != CS_ID_NONE) {
+        if (CutsceneManager_IsNext(this->picto.actor.csId)) {
+            CutsceneManager_StartWithPlayerCs(this->picto.actor.csId, &this->picto.actor);
+            Camera_SetFocalActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(this->picto.actor.csId)),
+                                 &this->picto.actor);
+        } else {
+            CutsceneManager_Queue(this->picto.actor.csId);
+        }
+    }
+
+    if (this->timer > 30) {
+        this->timer = 0;
+        EnBal_SetupFallFromSky(this);
+    } else {
+        if (this->timer > 10) {
+            // Freeze animation and induce blink
+            this->skelAnime.playSpeed = 0.0f;
+            if (this->eyeTimer > 6) {
+                this->eyeTimer = 6;
+            }
+            this->timer++;
+        } else {
+            this->timer++;
+        }
+    }
+}
+
+void EnBal_SetupFallFromSky(EnBal* this) {
```

Comment:
```suggestion
void EnBal_SetupFall(EnBal* this) {
```
Not sure `FromSky` adds much, same for the actual action function.